examples/five_by_five.py

```diff
@@ -0,0 +1,273 @@
+        """Compose the application screen."""
+        yield GameHeader()
+        yield GameGrid()
+        yield Footer()
+        yield WinnerMessage()
+
+    def toggle_cell(self, row: int, col: int) -> None:
+        """Toggle an individual cell, but only if it's on bounds.
+
+        :param int row: The row of the cell to toggle.
+        :param int col: The column of the cell to toggle.
+
+        If the row and column would place the cell out of bounds for the
+        game grid, this function call is a no-op. That is, it's safe to call
+        it with an invalid cell coordinate.
+        """
+        if 0 <= row <= (self.SIZE - 1) and 0 <= col <= (self.SIZE - 1):
+            self.cell(row, col).toggle_class("on")
+
+    _PATTERN: Final = (-1, 1, 0, 0, 0)
+
+    def toggle_cells(self, cell: GameCell) -> None:
+        """Toggle a 5x5 pattern around the given cell.
+
+        :param GameCell cell: The cell to toggle the cells around.
+        """
+        for row, col in zip(self._PATTERN, reversed(self._PATTERN)):
+            self.toggle_cell(cell.row + row, cell.col + col)
+        self.query_one(GameHeader).on = self.on_count
+
+    def make_move_on(self, cell: GameCell) -> None:
+        """Make a move on the given cell.
+
+        All relevant cells around the given cell are toggled as per the
+        game's rules.
+        """
+        self.toggle_cells(cell)
+        self.query_one(GameHeader).moves += 1
+        if self.all_on:
+            self.query_one(WinnerMessage).show(self.query_one(GameHeader).moves)
+            self.game_playable(False)
+
+    def on_button_pressed(self, event: GameCell.Pressed) -> None:
+        """React to a press of a button on the game grid."""
+        self.make_move_on(cast(GameCell, event.button))
+
+    def action_new_game(self) -> None:
+        """Start a new game."""
+        self.query_one(GameHeader).moves = 0
+        self.on_cells.remove_class("on")
+        self.query_one(WinnerMessage).hide()
+        middle = self.cell(self.SIZE // 2, self.SIZE // 2)
+        self.toggle_cells(middle)
+        self.set_focus(middle)
+        self.game_playable(True)
+
+    def action_navigate(self, row: int, col: int) -> None:
+        """Navigate to a new cell by the given offsets."""
+        if isinstance(self.focused, GameCell):
+            self.set_focus(
+                self.cell(
+                    (self.focused.row + row) % self.SIZE,
+                    (self.focused.col + col) % self.SIZE,
+                )
+            )
+
+    def action_move(self) -> None:
+        """Make a move on the current cell."""
+        if isinstance(self.focused, GameCell):
+            self.focused.press()
+
+    def on_mount(self) -> None:
+        """Get the game started when we first mount."""
+        self.action_new_game()
+
+
+class FiveByFive(App[None]):
+    """Main 5x5 application class."""
+
+    #: The name of the stylesheet for the app.
+    CSS_PATH = Path(__file__).with_suffix(".css")
```

Comment:
I think this can be `CSS_PATH="FiveByFive.css"` now.

Reply:
Changed.